proto/tendermint/abci/types.proto

```diff
@@ -412,8 +439,23 @@ message ValidatorUpdate {
 message VoteInfo {
   Validator validator         = 1 [(gogoproto.nullable) = false];
   bool      signed_last_block = 2;
-  reserved 3;  // Placeholder for tendermint_signed_extension in v0.37
-  reserved 4;  // Placeholder for app_signed_extension in v0.37
+}
+
+// ExtendedVoteInfo
+message ExtendedVoteInfo {
+  Validator validator         = 1 [(gogoproto.nullable) = false];
+  bool      signed_last_block = 2;
+  bytes     vote_extension    = 3;
+}
+
+// CanonicalVoteExtension
+// TODO: move this to core Tendermint data structures
```

Comment:
I'm curious why we wouldn't want this to be a standard part of the extension structure? Aren't all these properties inherent in the consensus mechanism?

Reply:
The `Canonical*` types are usually used to produce the bytes that Tendermint will sign over. Some of these fields aren't really relevant to the application during `PrepareProposal`/`FinalizeBlock` or would be duplicated by fields elsewhere in the request. They do need to be signed over though because they uniquely identify the vote extension and will be verified by _Tendermint_ but don't really need to be verified by the app.

Reply:
I guess that's what I'm trying to get at: If we always have to sign over these, and we don't know whether or not they're duplicated elsewhere, don't we basically have to include them regardless?

Reply:
Oh, I understand now. These are not used as the wire encodings of the data, but just as a package for the library API?

Reply:
It's not even part of the ABCI++ API (see the TODO above, line 452), it's part of the core data structures, just as `CanonicalVote`. I added it here with that TODO because I was too afraid of reaching beyond the ABCI++ files... before the whole design change is accepted.